main.cpp

```diff
@@ -51,23 +51,438 @@ If you need inspiration for what to write, take a look at previously approved st
+    return milesUntilOilChange;                                       
+}
+    
+bool Car::predictRaceWinProbability(std::string make, int maxSpeed)
+{
+    int competitorMaxSpeed = 200;
+
+    if (maxSpeed > competitorMaxSpeed)
+    {
+        std::cout << "You will most likely win in a race" << std::endl;
+        return true;
+    }
+
+    std::cout << "You will most likely lose in a race" << std::endl;
+    return false;
+    
+}
+
+Car::Vehicle::Vehicle():
+maxSpeed(160),
+turboCharged(false),
+vin("1BBHHK7772122340"),
+engineDisplacement(5.7f)
+{
+    std::cout << "Vehicle constructor called" << std::endl;
+}
+
+Car::Vehicle::~Vehicle()
+{
+    std::cout << "Vehicle Destructor called" << std::endl;
+}
+
+void Car::Vehicle::renovateVehicle(std::string vin, int maxSpeed, bool turboCharged)
+{
+    if((vin[9] < 'J') && (maxSpeed < 155))
+    {
+        std::cout << "Renovate vehicle" << std::endl;
+    }
+
+    std::cout << "Don't renovate vehicle" << std::endl;
+}
+
+float Car::Vehicle::calculateHpToTheWheel(float engineDisplacement, bool turboCharged, int maxSpeed)
+{
+    float horsepower;
+
+    if(turboCharged)
+    {
+        horsepower = ((engineDisplacement * 100) + 100 + maxSpeed);
+    }
+
+    else
+    {
+        horsepower = ((engineDisplacement * 100) + maxSpeed);
+    }
+
+    return horsepower;
+}
+
+void Car::Vehicle::resetEngineControlModule(std::string vin, bool vehicleIsImmobolized)
+{
+    if((vin[9] > 'J') && vehicleIsImmobolized)
+    {
+        std::cout << "Reset module exists and starting ECM reset" << std::endl;
+    }
+    std::cout << "Reset module does not exist" << std::endl;
+}
 
 /*
  UDT 2:
  */
+struct SoccerTeam
+{
+    int numberOfPlayers = 11;
+    float stoppageTime;
+    int numberOfGoalsCurrentGame;
+    std::string starPlayer;
 
+    void transferPlayerToTeam(std::string transferToTeam, std::string transferFromTeam);
+
+    double calculateWinProbability(float stoppageTime, int numberOfGoalsCurrentGame);
+
+    int calculateTotalSeasonGoals(int numberOfGoalsCurrentGame, int numberOfGoalsUntilNow);
+
+    SoccerTeam();
+
+    ~SoccerTeam();
+
+};
+
+SoccerTeam::SoccerTeam():
+stoppageTime(2.34f),
+numberOfGoalsCurrentGame(3),
+starPlayer("Neymar")
+{
+    std::cout << "SoccerTeam constructor called" << std::endl;
+};
+SoccerTeam::~SoccerTeam()
+{
+    std::cout << "SoccerTeam destructor called" << std::endl;
+}
+
+void SoccerTeam::transferPlayerToTeam(std::string transferToTeam, std::string transferFromTeam)
+{
+    std::cout << "Transferring from team " << transferFromTeam << " to team " << transferToTeam << std::endl;
+
+}
+
+
+double SoccerTeam::calculateWinProbability(float stoppageTime, int numberOfGoalsCurrentGame)
+{
+    if((stoppageTime < 3.0f) && (numberOfGoalsCurrentGame < 2))
+    {
+        std::cout << "Win probability over 50%" << std::endl;
+        return 0.6;
+    }
+
+    else
+    {
+        std::cout << "Win probability less than 50%" << std::endl;
+        return 0.4;
+    }
+
+}
```

Comment:
If you find yourself writing code like this:
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    else 
    {
        return anotherValue;
    }
}
```
You do not need the `else` block.  Here's why:
`return` statements are the last thing that **ever happens** in a function.
When the `if()` expression is evaluated as `true` and the block is entered, the `return aValue;` is the last line of code to be executed before the function exits.
but when the `if()` expression evaluates as `false` and the block is **NOT** entered, the next thing that happens is `return anotherValue;`
you have no code after the `else` block that needs executing, since your `else` block contains a `return` statement.
that means you do **NOT** need the `else` block, and it is just causing visual clutter.
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    
    return anotherValue;
}
```
- [ ] revise